tools/src/wyvern/tools/parsing/coreparser/arch/ASTComponentTypeDecl.java

```diff
@@ -0,0 +1,175 @@
+/* Generated By:JJTree: Do not edit this line. ASTComponentTypeDecl.java Version 4.3 */
+/* JavaCCOptions:MULTI=true,NODE_USES_PARSER=false,VISITOR=true,TRACK_TOKENS=false,NODE_PREFIX=AST,
+NODE_EXTENDS=,NODE_FACTORY=,SUPPORT_CLASS_VISIBILITY_PUBLIC=true */
+package wyvern.tools.parsing.coreparser.arch;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import wyvern.target.corewyvernIL.FormalArg;
+import wyvern.target.corewyvernIL.decl.Declaration;
+import wyvern.target.corewyvernIL.decl.DefDeclaration;
+import wyvern.target.corewyvernIL.decltype.DeclType;
+import wyvern.target.corewyvernIL.decltype.ValDeclType;
+import wyvern.target.corewyvernIL.expression.Expression;
+import wyvern.target.corewyvernIL.expression.New;
+import wyvern.target.corewyvernIL.modules.Module;
+import wyvern.target.corewyvernIL.support.InterpreterState;
+import wyvern.target.corewyvernIL.type.NominalType;
+import wyvern.target.corewyvernIL.type.StructuralType;
+import wyvern.tools.errors.ErrorMessage;
+import wyvern.tools.errors.ToolError;
+
+public class ASTComponentTypeDecl extends SimpleNode {
+    private boolean isExternal = false;
+    private String typeName;
+    private HashMap<String, String> reqs = new HashMap<>();
+    private HashMap<String, String> provs = new HashMap<>();
```

Comment:
Please add a comment to describe what the keys and values of these two hashmaps are.